apollo-idling-resource/src/main/java/com/apollographql/apollo/espresso/ApolloIdlingResource.java

```diff
@@ -0,0 +1,61 @@
+package com.apollographql.apollo.espresso;
+
+
+import android.support.test.espresso.IdlingResource;
+
+import com.apollographql.apollo.ApolloClient;
+import com.apollographql.apollo.IdleCallback;
+
+import javax.annotation.Nonnull;
+
+import static com.apollographql.apollo.api.internal.Utils.checkNotNull;
+
+/**
+ * An Espresso {@link IdlingResource} for {@link com.apollographql.apollo.ApolloClient}.
+ */
+public class ApolloIdlingResource implements IdlingResource {
```

Comment:
Making this class a part of any other module would have meant that users who don't want an idling resource would also end up getting one. So I felt it's better to keep it in a module of it's own, so that whoever wants an ApolloIdlingResource can simply pull in the dependency from maven.